front/src/modules/simulationResult/components/SpaceTimeChart/SpaceTimeChart.tsx

```diff
@@ -177,14 +180,16 @@ export default function SpaceTimeChart(props: SpaceTimeChartProps) {
     dragShiftTrain(dragOffset);
   }, [dragOffset]);
 
-  const redrawChart = () => {
+  const redrawChart = (newResizedChart?: Chart) => {
```

Comment:
The 3 times this function is called, we don't put any arguments. Do we need this new parameter ?

Reply:
You're right. I did this because I wanted to pass the new chart with the new updated scale here, but I forgot to do so and it still works. I think it's because I didn't cloneDeep chart.

I can resize directly the chart and remove this argument, but I'm a bit uncomfortable because we kind of update a state value without using the setter. That's why I tried to clone the chart and update the cloned chart.

What do you think ?

Reply:
Ok got it. I agree, it would be better not to update the state directly.

Reply:
I tried to remove this, but actually even if I cloneDeep, it seems to impact the global chart. So maybe it's better to leave the code like this